src/System.Diagnostics.Process/src/System/Diagnostics/ProcessManager.OSX.cs

```diff
@@ -18,40 +19,87 @@ public static int[] GetProcessIds()
 
         private static ProcessInfo CreateProcessInfo(int pid)
         {
-            ProcessInfo dummy = new ProcessInfo() // assign to fields to suppress warnings until implemented
+            // Negative PIDs aren't valid
+            if (pid < 0)
+            {
+                throw new ArgumentOutOfRangeException("pid");
+            }
+
+            ProcessInfo procInfo = new ProcessInfo();
+
+            // Try to get the task info. This can fail if the user permissions don't permit
+            // this user context to query the specified process
+            Interop.libproc.proc_taskallinfo? info = Interop.libproc.GetProcessInfoById(pid);
+            if (info.HasValue)
             {
-                _basePriority = 0,
-                _pageFileBytes = 0,
-                _handleCount = 0,
-                _pageFileBytesPeak = 0,
-                _poolNonpagedBytes = 0,
-                _poolPagedBytes = 0,
-                _privateBytes = 0,
-                _processId = pid,
-                _processName = null,
-                _sessionId = 0,
-                _virtualBytes = 0,
-                _virtualBytesPeak = 0,
-                _workingSet = 0,
-                _workingSetPeak = 0
-            };
-            dummy._threadInfoList.Add(new ThreadInfo()
+                // Set the values we have; all the other values don't have meaning or don't exist on OSX
+                procInfo.BasePriority = info.Value.ptinfo.pti_priority;
+                procInfo.HandleCount = Interop.libproc.GetFileDescriptorsForPid(pid).Length;
+                procInfo.ProcessId = pid;
+                procInfo.ProcessName = GetStringFromProcInfo(info.Value.pbsd).Trim('\0');
```

Comment:
@ellismg can correct me if I'm wrong, but I believe you could also just use Marshal.PtrToStringAnsi(info.Value.pbsd) and it would have the same result on OSX without needing a custom function.